src/services/api/trustedclients/trustedclients.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Injectable, HttpException, HttpStatus } from '@nestjs/common'
+import { JwtService } from '@nestjs/jwt'
+import AES from 'crypto-js/aes'
+import 'dotenv/config'
+
+@Injectable()
+export class TrustedclientsService {
+  constructor(private readonly jwtService: JwtService) {}
+
+  getEncodedMemo(client: string, username: string): string {
+    const clients = process.env.TRUSTED_CLIENTS.split(',')
+    const keys = process.env.TRUSTED_CLIENTS.split(',')
+    const indexOfClient = clients.indexOf(client)
+    if (indexOfClient <= -1) {
+      throw new HttpException(
+        `No such trusted client found with name - ${client}`,
+        HttpStatus.NOT_FOUND,
+      )
+    }
+    if (keys.length <= indexOfClient) {
+      throw new HttpException(
+        `Key not found for trusted client with name - ${client}`,
+        HttpStatus.NOT_FOUND,
+      )
+    }
+    const key = keys[indexOfClient]
+    var dataToSign = { userid: username, network: 'hive', banned: false }
+    var access_token = this.jwtService.sign(dataToSign)
+    var encrypted_access_token = AES.encrypt(access_token, key).toString()
+
+    /*
```

Comment:
BTW, I think we should be using DIDs for identifying trusted clients. So for each trusted client they have will have key-did-ed25519 public key where we can encrypt messages to. Along with them being able to sign a verifiable message as well.